src/frequenz/client/microgrid/_client.py

```diff
@@ -261,20 +261,21 @@ async def _new_component_data_receiver(
             expected_category,
         )
 
-        broadcaster = self._broadcasters.setdefault(
-            component_id,
-            streaming.GrpcStreamBroadcaster(
+        broadcaster = self._broadcasters.get(component_id)
+        if broadcaster is None:
+            broadcaster = streaming.GrpcStreamBroadcaster(
                 f"raw-component-data-{component_id}",
                 # We need to cast here because grpc says StreamComponentData is
                 # a grpc.CallIterator[PbComponentData], not a
                 # grpc.aio.UnaryStreamCall[..., PbComponentData].
                 lambda: cast(
-                    grpc.aio.UnaryStreamCall[Any, PbComponentData],
+                    AsyncIterator[PbComponentData],
```

Comment:
The comments above seems outdated with the current changes

Reply:
Ah, yeah, thanks! Will update!